src/content/platform-integration/web/embedding-flutter-web.md

```diff
@@ -0,0 +1,326 @@
+attached. This `viewId` can be used to uniquely identify each view, retrieve
+its initial configuration, or decide what to render in it.
+
+The `viewId` of the `FlutterView` that is being rendered can be retrieved from
+its `BuildContext` like this:
+
+```dart highlightLines=4-5
+class SomeWidget extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    // Retrieve the `viewId` where this Widget is being built:
+    final int viewId = View.of(context).viewId;
+    // ...
+```
+
+Read more about the [`View.of` constructor][].
+
+[`View.of` constructor]: {{site.api}}/flutter/widgets/View/of.html
+
+### Initial view configuration
+
+Flutter views can receive any initialization data from JS when starting up.
+The values are passed through the `initialData` property of the `addView`
+method, as shown:
+
+```js highlightLines=4-7
+// Adding a view with initial data...
+let viewId = app.addView({
+  hostElement: someElement,
+  initialData: {
+    greeting: 'Hello, world!',
+    randomValue: Math.floor(Math.random() * 100),
+  }
+});
+```
+
+In Dart, the `initialData` is available as a `JSAny` object, accessible through
+the top-level `views` property in the `dart:ui_web` library. The data is
+accessed through the `viewId` of the current view, like so: 
+
+```dart
+final initialData = ui_web.views.getInitialData(viewId) as YourJsInteropType;
+```
+
+To learn how to define the `YourJsInteropType` class to map the `initialData`
+object passed from JS so it's type-safe in your Dart program, check out:
+[JS Interoperability][] on dart.dev.
+
+[JS Interoperability]: {{site.dart-site}}/interop/js-interop
+
+### View constraints
+
+By default, an embedded Flutter web view considers the size of its `hostElement`
+as an immutable property, and will tightly constrain its layout to the available
+space.
+
+On the web, it's common for the intrinsic size of an element to affect the
+layout of the page (like an `img` tag or `p` tags that can reflow content around
+them).
+
+When adding a view to Flutter web, you may configure it with constraints that
+inform Flutter of how the view needs to be laid out:
+
+```js highlightLines=4-8
+// Adding a view with initial data...
+let viewId = app.addView({
+  hostElement: someElement,
+  viewConstraints: {
+    maxWidth: 320,
+    minHeight: 0,
+    maxHeight: Infinity,
+  }
+});
+```
+
+The view constraints passed from JS need to be compatible with the CSS styling
+of the `hostElement` where Flutter is being embedded. For example, Flutter
+won’t try to "fix" contradictory constants like passing  `max-height: 100px`
+in CSS, but `maxHeight: Infinity` to Flutter.
+
+Learn more about the [`ViewConstraints` class][], and
+[Understanding constraints][].
+
+[`ViewConstraints` class]: {{site.api}}/flutter/dart-ui/ViewConstraints-class.html
+[Understanding constraints]: {{site.docs}}/ui/layout/constraints
+
+## Custom element (`hostElement`)
+
+_Between Flutter 3.10 and 3.24_<br />
```

Comment:
Will this be true in 3.25? If so, we should say "In Flutter 3.10 and beyond..."

Reply:
This will be still true, but starting in 3.24 people should prefer the other embedded method

Reply:
I see! OK, I'm going to make another suggested change.